src/openrct2/world/Footpath.cpp

```diff
@@ -1279,39 +1273,39 @@ static bool get_next_direction(int32_t edges, int32_t* direction)
  *              (1 << 7): Ignore no entry signs
  */
 static int32_t footpath_is_connected_to_map_edge_recurse(
-    int32_t x, int32_t y, int32_t z, int32_t direction, int32_t flags, int32_t level, int32_t distanceFromJunction,
-    int32_t junctionTolerance)
+    const CoordsXYZD& footpathPos, int32_t flags, int32_t level, int32_t distanceFromJunction, int32_t junctionTolerance)
```

Comment:
I'm not convinced that direction should be part of footpathPos. It is used independently and represents the targetDirection and not the `footpathPos`